src/Tree/HydroTree.cpp

```diff
@@ -686,11 +710,37 @@ void HydroTree<ndim,ParticleType>::UpdateTimestepsLimitsFromDistantParticles
       if (Nactive == 0) continue ;
 
       // Make local copies of active particles & update vsig_max
-      for (j=0; j<Nactive; j++) activepart[j] = partdata[activelist[j]];
+      for (int j=0; j<Nactive; j++) activepart[j] = partdata[activelist[j]];
 
       tree->ComputeSignalVelocityFromDistantInteractions(cell, Nactive, activepart, partdata);
 
-      for (j=0; j<Nactive; j++) {
+#ifdef MPI_PARALLEL
+      if (!only_imported_particles) {
+		  // Loop over pruned trees as well
+		  // Doing it after the local tree walk is more efficient as vsig_max might have been already increased
+		  // TODO: is it more efficient to have this loop inside or outside the one over particles? In this way we are trashing
```

Comment:
First off, I think it's great that this is working and I suspect that the possible performance gains here might be small.  However, perhaps it is worth thinking about this more generally.

The code would be easier to understand if we seperated out the look computing the local criterion and the pruned tree walk because we would be able to get rid of a bunch of if(!onlly...) statements making it easier to follow the logic. I'm in favour based on those grounds, along with the fact that I doubt we will see much of a performance difference.

As far as maintaining cache efficiency and thread utilization: would the openMP collapse(2) directive help?